vars/groups.groovy

```diff
@@ -4,40 +4,24 @@ import groovy.json.JsonOutput
 def createGroup(String groupName, List groupMembers, String groupType) {
    return JsonOutput.toJson([
       name: groupName,
-      members: groupMembers,
+      discordID: groupDiscordID,
+      swarmID: groupSwarmID,
       type: groupType
    ]);
 }
 
-def getGroupMembers(groups, String groupName) {
-    for(group in groups) {
-        if(group.name == groupName) {
-            return group.members;
-        }
-    }
-
-    log.error("Tried to get group members for group '${groupName}', but this group couldn't be found!");
-    return [];
-}
-
-def getGroupType(groups, String groupName) {
-    for(group in groups) {
-        if(group.name == groupName) {
-            return group.type;
-        }
-    }
-
-    log.error("Tried to get group type for group '${groupName}', but this group couldn't be found!");
-    return [];
-}
-
 def mentionGroup(groups, String groupName) {
-    def groupMembers = getGroupMembers(groups, groupName);
-    def groupType = getGroupType(groups, groupName);
-
-    if(!groupMembers && !groupType) {
-        return '';
-    }
+   def groupType = ""
+   def discordID = ""
+   def groupsParsed = new JsonSlurper().parseText(groups)
+
+   groupsParsed.groups.each { group ->
+      if (group.name == groupName)
+      {
+         groupType = group.type
+         discordID = group.discordID
+      }
+   }
```

Comment:
Is it possible to break out of this loop prematurely?

Reply:
Breaking out of for-each loops in Groovy is illegal ([https://stackoverflow.com/questions/3049790/can-you-break-from-a-groovy-each-closure]). However using find ([https://www.tutorialspoint.com/groovy/groovy_find.htm]) we can break out of the loop prematurely using return true.